From a9e60d40c1780022112ae9e1cd964a3648bb7d47 Mon Sep 17 00:00:00 2001
From: Ben Boeckel <ben.boeckel@kitware.com>
Date: Sat, 18 Jan 2025 01:13:37 +0100
Subject: [PATCH] IO/FFMPEG: avoid deprecated `avcodec_close`

It hasn't been necessary for a long time, so mask its use.
---
 IO/FFMPEG/vtkFFMPEGVideoSource.cxx | 6 ++++++
 IO/FFMPEG/vtkFFMPEGWriter.cxx      | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/IO/FFMPEG/vtkFFMPEGVideoSource.cxx b/IO/FFMPEG/vtkFFMPEGVideoSource.cxx
index f2366e7152a..c7e5c93d91a 100644
--- a/IO/FFMPEG/vtkFFMPEGVideoSource.cxx
+++ b/IO/FFMPEG/vtkFFMPEGVideoSource.cxx
@@ -72,12 +72,18 @@ class vtkFFMPEGVideoSourceInternal
     }
     if (this->VideoDecodeContext)
     {
+#if defined(LIBAVCODEC_VERSION_MAJOR) && LIBAVCODEC_VERSION_MAJOR < 62
       avcodec_close(this->VideoDecodeContext);
+#endif
+      avcodec_free_context(&this->VideoDecodeContext);
       this->VideoDecodeContext = nullptr;
     }
     if (this->AudioDecodeContext)
     {
+#if defined(LIBAVCODEC_VERSION_MAJOR) && LIBAVCODEC_VERSION_MAJOR < 62
       avcodec_close(this->AudioDecodeContext);
+#endif
+      avcodec_free_context(&this->AudioDecodeContext);
       this->AudioDecodeContext = nullptr;
     }
     if (this->FormatContext)
diff --git a/IO/FFMPEG/vtkFFMPEGWriter.cxx b/IO/FFMPEG/vtkFFMPEGWriter.cxx
index 2cb8083fef7..cdf8c05e21f 100644
--- a/IO/FFMPEG/vtkFFMPEGWriter.cxx
+++ b/IO/FFMPEG/vtkFFMPEGWriter.cxx
@@ -363,7 +363,9 @@ void vtkFFMPEGWriterInternal::End()
 
   if (this->avCodecContext)
   {
+#if defined(LIBAVCODEC_VERSION_MAJOR) && LIBAVCODEC_VERSION_MAJOR < 62
     avcodec_close(this->avCodecContext);
+#endif
     avcodec_free_context(&this->avCodecContext);
     this->avCodecContext = nullptr;
   }
